packages/es-components/src/components/controls/textbox/Textbox.specs.js

```diff
@@ -1,208 +1,48 @@
 /* eslint-env jest */
 
 import React from 'react';
-import { mountWithTheme, renderWithTheme } from 'styled-enzyme';
+import { render, fireEvent, cleanup } from 'react-testing-library';
+import { ThemeProvider } from 'styled-components';
+import viaTheme from 'es-components-via-theme';
 
-import Icon from '../../base/icons/Icon';
 import Textbox from './Textbox';
 
-jest.mock('../../util/generateAlphaName', () => () => 'abcdef');
-
-const buildTextbox = props => <Textbox labelText="Text" {...props} />;
-
-describe('Textbox component', () => {
-  const handleOnChange = jest.fn();
-  const handleOnBlur = jest.fn();
-  const mockEvent = { target: { value: '112' } };
-
-  it('renders as expected', () => {
-    const props = {
-      onChange: handleOnChange,
-      value: 'testvalue',
-      id: 'test-id',
-      name: 'test-textbox'
-    };
-    const tree = renderWithTheme(buildTextbox(props)).toJSON();
-    expect(tree).toMatchSnapshot();
-  });
-
-  it('generates id if no id is provided', () => {
-    const props = {
-      onChange: handleOnChange,
-      value: 'testvalue'
-    };
-    const tree = renderWithTheme(buildTextbox(props)).toJSON();
-    expect(tree).toMatchSnapshot();
-  });
-
-  it('includes additional text when labelSuffix is provided', () => {
-    const props = {
-      onChange: handleOnChange,
-      value: 'testvalue',
-      labelSuffix: <span>Optional</span>
-    };
-    const tree = renderWithTheme(buildTextbox(props)).toJSON();
-    expect(tree).toMatchSnapshot();
-  });
-
-  it('executes the handleOnChange function when text is changed', () => {
-    const props = {
-      onChange: handleOnChange
-    };
-    const input = mountWithTheme(buildTextbox(props)).find('input');
-
-    input.simulate('change', mockEvent);
-    expect(handleOnChange).toBeCalled();
-    handleOnChange.mockClear();
-  });
-
-  it('executes the handeOnBlur function when the input focus is lost', () => {
-    const props = {
-      onBlur: handleOnBlur
-    };
-    const input = mountWithTheme(buildTextbox(props)).find('input');
-
-    input.simulate('blur', mockEvent);
-    expect(handleOnBlur).toBeCalled();
-    handleOnBlur.mockClear();
-  });
-
-  it('renders addon icon when "prependIconName" prop is passed', () => {
-    const props = {
-      prependIconName: 'prepend'
-    };
-    const instance = mountWithTheme(buildTextbox(props));
-
-    expect(instance.find(Icon).length).toBe(1);
-  });
-
-  it('renders addon icon when "appendIconName" prop is passed', () => {
-    const props = {
-      appendIconName: 'append'
-    };
-    const instance = mountWithTheme(buildTextbox(props));
-
-    expect(instance.find(Icon).length).toBe(1);
-  });
-
-  it('renders addon icons when both "prependIconName" and "appendIconName" props are passed', () => {
```

Comment:
There were a lot of tests removed from Textbox, any plans to add more? Several of these would probably be good candidates to just throw in a snapshot (like checking prepend icons).

Reply:
I added a few more tests. I think I was just being a little lazy. 😞